docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+
+## Guiding principles
+
+The main idea behind contributions that get merged into SoH's main repo is well-built, easy to maintain changes that most often modify, extend, or improve built-in functionality. Things like custom items, for example, conflict with vanilla, and thus should remain as custom builds. However, you never know what might be considered for merging, so feel free to PR, or you can talk to devs about your idea on the [Discord](https://discord.gg/shipofharkinian).
+
+## Repo management
+
+In addition to principles outlined in MODDING.md, the importance of maintaining clean copies of the main repos (develop, and develop-{release}) cannot be overstated. It is easier to create new branches and update existing ones with changes from SoH using branches on your fork that are always exact mirrors of the primary branches on HM64's repo. All changes should be committed to new branches on your own repo, and those branches form the base of PRs you then submit to HM64.
+
+## Formatting
+
+While we are still in the process of codifying code formats, whitespacing, etc, generally you can follow the precedents set around where you're making changes for information on how to indent lines, opening bracket positions, capitalization, variable naming, etc. Be specific, be as concisely elaborate/accurate as you can.  For example:
+
+```c++
+    if (CVarGetInteger("gCustomizeFrogsOcarinaGame", 0)) {
+        this->frogSongTimer = sTimerFrogSong[index] * CVarGetInteger("gFrogsModifyFailTime", 1);
+    } else {
+        this->frogSongTimer = sTimerFrogSong[index];
+    }
+```
+
+Here you see we open brackets on the same line, spacing outside the parentheses but not inside, same line else and opening bracket, etc. Also, all if...else blocks should be fully encapsulated with brackets, even if they're single-line blocks (which technically would be legal code).
+
+## Documentation
+
+We want to maintain the readability and integrity of vanilla codeflow as much as possible going forward. To do this, we have decided that we will want any modifications to vanilla code to be well notated and maintained separately from modified sections. Given this original code:
+
+```c++
+        this->frogSongTimer = 40;
+        this->ocarinaNoteIndex = 0;
+        func_8010BD58(play, OCARINA_ACTION_FROGS);
+        this->ocarinaNote = EnFr_GetNextNoteFrogSong(this->ocarinaNoteIndex);
+        EnFr_CheckOcarinaInputFrogSong(this->ocarinaNote);
+        this->actionFunc = EnFr_ContinueFrogSong;
+```
+
+If you wanted to modify this block with an enhancement to add difficulty options to the frog ocarina game, the first thing you would do is encapsulate it in a comment region as such:
+
+```c++
+        //Enhancement: Customize Frogs Ocarina Game
```

Comment:
Personally I think a well named cvar and/or function is enough to not have comments like this. But if we must let’s match or augment the existing patterns I’ve been using with code regions

Reply:
Got an example off the top of your head?

Reply:
If you do a global search for `// #region` you can see the ones I have added, with a bunch more anchor related ones on my anchor branch

Reply:
So how would that apply to this instance? just a region around enhancement version, or would it still encapsulate the whole thing and have the `// vanilla code` comment?

Reply:
Well in this case, along with what bria mentioned above I think an aptly named function would suffice (`SoH_EnFr_SetupFrogSong`,  or `Enhancements_EnFr_SetupFrogSong`) but if we wanted a comment as well yeah it would be wrapping around our version with
```c
// #region SOH [Enhancements] Customize Frogs Ocarina Game
...
// #endregion
```

Reply:
Right, but this example comes from an active PR, and I'm trying to figure out how far I want to take what I tell Spoon to do with it (as well as all the other ones he has that he already applied this to XD).

Reply:
I would say since this document and some of these practices are still in discussion/development I wouldn't press too hard for it to be followed. But in this particular example I'd just replace the first two comments with the ones I gave above, and remove the `// Enhancement end` comment

Reply:
So more like this?
```c++
        // #region SOH [Enhancement]
        if (CVarGetInteger("gCustomizeFrogsOcarinaGame", 0) == 1) {
            this->frogSongTimer = 40 * CVarGetInteger("gFrogsModifyFailTime", 1);
            if (CVarGetInteger("gInstantFrogsGameWin", 0) == 1) {
                this->actor.textId = 0x40AC;
                EnFr_SetupReward(this, play, false);
            } else {
                this->ocarinaNoteIndex = 0;
                func_8010BD58(play, OCARINA_ACTION_FROGS);
                this->ocarinaNote = EnFr_GetNextNoteFrogSong(this->ocarinaNoteIndex);
                EnFr_CheckOcarinaInputFrogSong(this->ocarinaNote);
                this->actionFunc = EnFr_ContinueFrogSong;
            }
        } else { // #endregion
            this->frogSongTimer = 40;
            this->ocarinaNoteIndex = 0;
            func_8010BD58(play, OCARINA_ACTION_FROGS);
            this->ocarinaNote = EnFr_GetNextNoteFrogSong(this->ocarinaNoteIndex);
            EnFr_CheckOcarinaInputFrogSong(this->ocarinaNote);
            this->actionFunc = EnFr_ContinueFrogSong;
        }
```

Reply:
Yeah, I'm not sure if VS/VSCode properly folds if unless #endregion is on it's own line or not, but that working isn't as important.

Also I'd say people are free to put whatever after the `[Enhancement]` part if they want, or just leave it blank and let the code/cvar tell the story